csharp/src/Drivers/Apache/Hive2/DecimalUtility.cs

```diff
@@ -83,33 +86,27 @@ private static void FillBytes(Span<byte> bytes, BigInteger integerValue, int byt
             }
         }
 
-        private static BigInteger ToBigInteger(string value, int precision, int scale)
+        private static BigInteger ToBigInteger(ReadOnlySpan<byte> value, int precision, int scale)
         {
-            BigInteger integerValue;
-#if NETCOREAPP
-            ReadOnlySpan<char> significantValue = GetSignificantValue(value, precision, scale);
-            integerValue = BigInteger.Parse(significantValue);
-#else
-            ReadOnlySpan<char> significantValue = GetSignificantValue(value.AsSpan(), precision, scale);
-            integerValue = BigInteger.Parse(significantValue.ToString());
-#endif
+            ReadOnlySpan<byte> significantValue = GetSignificantValue(value, precision, scale);
+            BigInteger integerValue = BigInteger.Parse(Encoding.UTF8.GetString(significantValue));
```

Comment:
NETCOREAPP could probably stackalloc a char buffer and avoid the heap allocation, but I don't know if that's worthwhile.
```
            int len = Encoding.UTF8.GetCharCount(significantValue);
            ReadOnlySpan<char> chars = stackalloc char[significantValue.Length];
            Encoding.UTF8.GetChars(significantValue, chars);
            BigInteger.Parse(chars);
```
(It's even simpler if we know that the bytes are all code points < 128.)

Reply:
Updated with your suggestions.